sdk/python/kfp/components/structures_test.py

```diff
@@ -511,6 +513,23 @@ def test_validate_command_and_arg(self):
             lambda: structures.ContainerSpec(
                 image='python:3.7', command=[], args=commands))
 
+    def test_raise_error_if_access_artifact_by_itself(self):
+
+        def comp_with_artifact_input(dataset: dsl.Input[dsl.Dataset]):
```

Comment:
Should this have the container_component decorator?

Reply:
If the test is written as it is now, the `create_component_from_func` method should take in an unwrapped func, otherwise the error will be thrown before captured by pytest. An alternative way is to test against the entire container_component authoring, but I think the current way of writing makes it more clear where the error happens.

Reply:
I see. What you have now looks good.